tests/test_deserialization.py

```diff
@@ -1178,6 +1178,18 @@ class AliasingUserSerializer(Schema):
         assert result["email"] == "foo@bar.com"
         assert "years" not in result
 
+    def test_deserialize_dotted_path_with_data_key_param(self):
+        class MySchema(Schema):
+            foo = fields.Str(data_key="foo")
```

Comment:
`data_key` is equal to field name

Reply:
Good point it is redundant here, I simply wanted the test to include a case with a dotted path key and without. 

I can make a small change her to this data key so it is different, and a change so the non dotted path field doesn't have the word "dotted" in the test case example as per your comment below! I can probably get this done some time tomorrow.

Reply:
Good.

> I can make a small change her to this data key so it is different

Yes. But this is already tested in the test above. However, maybe you can test a multiple dot path like "data.foo/bar".